### ISSUE TITLE
Using dorado trim to remove primers from both the end

### ISSUE THREAD

**Issue by ashokpatowary** [nanoporetech/dorado#1074]
Hi

I have a basic question about using `dorado trim`. I have sequences with different primers on both ends. I want to retain only the trimmed sequences where each primer is present on its respective end. To use `dorado trim`, should I provide both primer sequences in one file, or should I provide one primer sequence at a time?

Thanks

**Comment by malton-ont** [comment]
Hi @ashokpatowary,

You can provide multiple primer sequences in a single file. This should then trim any occurrence of either primer.

**Comment by ashokpatowary** [comment]
Hu @malton-ont 

Thanks for your response. I have a follow up question. Consider I should have 5' and 3' primer sequences in all my reads. How dorado trip will output if
1. the input sequence has only 5' primer. Will it trim the 5' and print the sequence though 3' primer is not present.
2. the input sequence does not have any of the 5' and 3' prime sequence. Will it print the input sequence as it in the output sequence though no primer sequence was present?

Thanks

**Comment by malton-ont** [comment]
@ashokpatowary,

`dorado trim` removes primers and adapters if they are present. If they are not, it leaves the sequence unchanged.

**Comment by malton-ont** [comment]
@nbat64,

Yes, a fasta file with multiple sequences is fine. See [the README](https://github.com/nanoporetech/dorado/blob/release-v0.8/README.md#custom-primer-trimming) for more info.
